shared/javascripts/context_messenger.js

```diff
@@ -316,43 +321,66 @@ if (Privly === undefined) {
 
   /** @inheritdoc 
    *
-   * "Workers" here refer to the Jetpack messaging API objects, which could 
-   * be workers created by page-mod, or a panel object or self. All these objects 
-   * provide the interface to send and receive messages using "port".
+   * "Workers" here refer to the dynamically created Jetpack messaging API objects, 
+   * which could be workers(Content Script instances) created by page-mod, or a panel object. 
+   * All these objects provide the interface to send and receive messages using "port".
+   * These objects, when created, should be added to the workers list/array using 
+   * Privly.message.currentAdapter.addWorker(<worker object>).
    *
-   * Stores all the workers Or messaging API objects.
-   * @see FirefoxAdapter.prototype.pushWorker
+   * Stores all the dynamically created workers Or messaging API objects.
    */
   FirefoxAdapter.prototype.workers = [];
 
   /** @inheritdoc
    * Adds a worker to the list of workers and listens for messages.
    *
-   * @param {Object} worker Jetpack messaging API object
+   * @param {Object} worker Jetpack messaging API object.
    */
-  FirefoxAdapter.prototype.pushWorker = function(worker) {
-    // Add worker to array of workers
-    this.workers.push(worker);
+  FirefoxAdapter.prototype.addWorker = function(worker) {
+   
+    var pushWorker = this.pushWorker.bind(this);
+    var popWorker = this.popWorker.bind(this);
     var callback = this.listener;
-    // Use the worker to listen for messages.
-    worker.port.on("privly", callback);
+    
+    // Add worker to array of workers
+    pushWorker(worker);
+    // Listen for messages.
+    worker.port.on("PRIVLY_MESSAGE", callback);
+    // the workers array should contain only active workers.
+    // remove the workers, that are inactive or destroyed, from the workers array.
+    worker.on("pageshow", function() { pushWorker(this); });
+    worker.on("pagehide", function() { popWorker(this); });
```

Comment:
Nope, it'll work. The background script doesn't message the content scripts to change the current setting. It instead injects new content scripts in all tabs passing the setting as a content script option, based on which the DOM element is inserted/changed. `privly.js` looks at the DOM element and then decides if it needs to run or not.